### PR TITLE
Add AotAutogradFallbackTests to dynamic suite

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100454

cc @soumith @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -3109,6 +3109,25 @@ def nan_to_num(self, nan=None, posinf=None, neginf=None):
...
+def t_(self):
+    ndims = self.ndim
+    assert ndims <= 2
+    size = self.size()
+
+    dim0 = size[0]
+    dim1 = size[1]
+
+    if dim0 == dim1:
+        return self
```
Comment (ezyang) [inline torch/_meta_registrations.py:3122]: This looks wrong. Where did you get this implementation from?

**Review by ezyang — CHANGES_REQUESTED** [review]
t_ implementation is wrong

File: torch/_meta_registrations.py (hunk 1 above)
Comment (voznesenskym) [reply torch/_meta_registrations.py:3122]: transpose_ in TensorShape.cpp

Comment (voznesenskym) [reply torch/_meta_registrations.py:3122]: Pretty new to writing meta registrations, lmk what's wrong here, and if there's a better place to check / look.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**